pkg/appauth/manager/json/json.go

```diff
@@ -199,20 +214,26 @@ func (mgr *jsonManager) GetAppPassword(ctx context.Context, userID *userpb.UserI
 		return nil, errtypes.NotFound("password not found")
 	}
 
-	pw, ok := appPassword[password]
-	if !ok {
-		return nil, errtypes.NotFound("password not found")
-	}
-
-	if pw.Expiration != nil && pw.Expiration.Seconds != 0 && uint64(time.Now().Unix()) > pw.Expiration.Seconds {
-		return nil, errtypes.NotFound("password not found")
+	for hash, pw := range appPassword {
+		err := bcrypt.CompareHashAndPassword([]byte(hash), []byte(password))
```

Comment:
Can we generate the hash of the password received from the user and look that up in the map? I'm not sure that'll work so look into that

Reply:
Nope, because the hash change each time you generate it, so the only way is to use the CompareHashAndPassword function

Reply:
The whole concept behind hashes is that you get the same hash for the same object every time

Reply:
I know, but apparently not with this lib :)